frontend/src/metabase/query_builder/components/chart-type-selector/use-chart-type-visualizations.ts

```diff
@@ -0,0 +1,94 @@
+import { useCallback, useMemo } from "react";
+import _ from "underscore";
+
+import visualizations from "metabase/visualizations";
+import { sanatizeResultData } from "metabase/visualizations/shared/utils/data";
+import type Question from "metabase-lib/v1/Question";
+import {
+  type CardDisplayType,
+  type Dataset,
+  type VisualizationDisplay,
+  isCardDisplayType,
+} from "metabase-types/api";
+
+import { DEFAULT_ORDER } from "./viz-order";
+
+export type UseChartTypeVisualizationsProps = {
+  question: Question;
+  onUpdateQuestion: (question: Question) => void;
+};
+
+export const useChartTypeVisualizations = ({
```

Comment:
This hook takes in the `question` and `result` and gives us the current visualization, the most sensible viz types, the other viz types, and a way to change the viz type. We can use this hook in the the SDK.